xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-ui/src/main/resources/XWiki/Lightbox/LightboxAdministration.xml

```diff
@@ -0,0 +1,349 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="XWiki.Lightbox.LightboxAdministration" locale="">
+  <web>XWiki.Lightbox</web>
+  <name>LightboxAdministration</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>LightboxAdministration</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity output="false"}}
+## To avoid mixing wiki syntax with html, unwrap the result of the xproperty displayer.
+#macro (stripHTMLMacro $displayOutput)
+  $stringtool.removeEnd($stringtool.removeStart($displayOutput, '{{html clean="false" wiki="false"}}'), '{{/html}}')
+#end
+#macro (showProperty $propertyName)
+  &lt;dl&gt;
+    &lt;dt&gt;
+      #set ($propertyClass = $xclass.get($propertyName))
+      &lt;label#if ($editing) for="${lightboxConfigClassName}_0_${propertyName}"#end&gt;
+        $escapetool.xml($propertyClass.translatedPrettyName)
+      &lt;/label&gt;
+      &lt;span class="xHint"&gt;
+        $escapetool.xml($services.localization.render($propertyClass.hint))
+      &lt;/span&gt;
+    &lt;/dt&gt;
+    #set ($displayOutput = $lightboxConfigDoc.display($propertyName, $mode))
+    &lt;dd&gt;#stripHTMLMacro($displayOutput)&lt;/dd&gt;
+  &lt;/dl&gt;
+#end
+{{/velocity}}
+
+{{velocity}}
+#set ($lightboxConfigClassName = 'XWiki.Lightbox.LightboxConfigurationClass')
+#set ($lightboxConfigDoc = $xwiki.getDocument('XWiki.Lightbox.LightboxConfiguration'))
+#set ($discard = $lightboxConfigDoc.use($lightboxConfigClassName))
+#set ($lightboxConfigClass = $xwiki.getDocument($lightboxConfigClassName))
+#set ($xclass = $lightboxConfigClass.getxWikiClass())
+#set ($editing = $xcontext.action == 'edit' || $xcontext.action == 'admin')
+#set ($mode = "#if ($editing)edit#{else}view#end")
+{{html clean="false"}}
+#if ($xcontext.action == 'admin')
+  &lt;form action="$lightboxConfigDoc.getURL('save')" class="xform"&gt;
+    &lt;div class="hidden"&gt;
+      &lt;input type="hidden" name="form_token" value="$!escapetool.xml($services.csrf.token)"/&gt;
+    &lt;/div&gt;
+#else
+  &lt;div class="xform"&gt;
+#end
+  #showProperty('isLightboxEnabled')
+#if ($xcontext.action == 'admin')
+  &lt;p&gt;
+    &lt;input type="submit" class="btn btn-primary" name="action_saveandcontinue"
+      value="$escapetool.xml($services.localization.render('save'))"/&gt;
+  &lt;/p&gt;
+&lt;/form&gt;
+#else
+&lt;/div&gt;
+#end
+{{/html}}
+{{/velocity}}</content>
```

Comment:
Maybe I'm missing something but not sure why you need that code: you could just use the `configurationClass` property of the ConfigurableClass to have automatically the administration UI based on the LightboxConfigurationClass, AFAICS it only consists in one boolean. Would be less complex and better for extending it in the future (no need to maintain this code). See: https://extensions.xwiki.org/xwiki/bin/view/Extension/Administration+Application#HMakingyourapplicationconfigurablewithConfigurableClass

Reply:
Indeed, it would simplify it. Thanks!